### ISSUE TITLE
sprintf / String::Formatter does not correctly implement fixed-width zero-extended "%.<width><letter>" format strings

### ISSUE THREAD

**Issue by postmodern** [crystal-lang/crystal#11255]
### Description

The C format string syntax allows specifying a fixed-width for the resulting strings, which will be zero-extended to match the desired width:

* `"%.64b"`
* `"%.22o"`
* `"%.20d"`
* `"%.16x"`

However, Crystal's implementation of `String::Formatter` does not seem to honor the fixed-width specifier and does not zero-extend the formatted Strings:

### Test Program

```crystal
num = 0xffffffff_i64

def debug_sprintf(fmt : String,value : Int)
  string = sprintf(fmt,value)
  puts "#{fmt}\t#{string}\t(size = #{string.size})"
end

debug_sprintf("%.64b",num)
debug_sprintf("%.22o",num)
debug_sprintf("%.20d",num)
debug_sprintf("%.16x",num)
```

### Expected Output

```
%.64b	0000000000000000000000000000000011111111111111111111111111111111	(size = 64)
%.22o	0000000000037777777777	(size = 22)
%.20d	00000000004294967295	(size = 20)
%.16x	00000000ffffffff	(size = 16)
```

### Actual Output

```
%.64b	11111111111111111111111111111111	(size = 32)
%.22o	37777777777	(size = 11)
%.20d	4294967295	(size = 10)
%.16x	ffffffff	(size = 8)
```

**Comment by naqvis** [comment]
it does honor that, but format is bit different.
```crystal
debug_sprintf("%064.b",num)
debug_sprintf("%022.o",num)
debug_sprintf("%020.d",num)
debug_sprintf("%016.x",num)
```
will print your expected output.

```
%064.b	0000000000000000000000000000000011111111111111111111111111111111	(size = 64)
%022.o	0000000000037777777777	(size = 22)
%020.d	00000000004294967295	(size = 20)
%016.x	00000000ffffffff	(size = 16)

```

**Comment by asterite** [comment]
It's definitely a bug. I copied Ruby here. If it doesn't work like Ruby, it's a bug.

**Comment by HertzDevil** [comment]
Should be fixed by #10973.
